tests/proxy_test.go

```diff
@@ -92,6 +92,41 @@ func TestBasicProxy_GRPC(t *testing.T) {
 	}
 }
 
+func TestProxyHandleError_GRPC(t *testing.T) {
+	invalidServer := httptest.NewUnstartedServer(newEchoServer("hello"))
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	proxy, cleanup, err := runGRPCProxyServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+
+	runAgent(proxy.agent, stopCh)
+
+	// Wait for agent to register on proxy server
+	time.Sleep(time.Second)
+
+	// run test client
+	tunnel, err := client.CreateGrpcTunnel(proxy.front, grpc.WithInsecure())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	c := &http.Client{
+		Transport: &http.Transport{
+			Dial: tunnel.Dial,
+		},
+	}
+
+	_, err = c.Get(invalidServer.URL)
+	if err == nil {
+		t.Error("Expected error when destination is unreachable, did not receive error")
```

Comment:
Check the error message?

Reply:
Just for clarification, do you mean compare the error message with a predefined string instead of just checking for nil?

Reply:
Right.

Reply:
httptest generates a random port for the server, and the error message is of the form: `Get "http://127.0.0.1:46415": dial tcp 127.0.0.1:46415: connect: connection refused`. I checked for the existence of `connection refused` in the error message instead.